ext/ffi/lib.rs

```diff
@@ -507,6 +560,177 @@ async fn op_ffi_call_nonblocking(
     .unwrap()
 }
 
+fn op_ffi_ptr_of<FP>(
+  state: &mut deno_core::OpState,
+  buf: ZeroCopyBuf,
+  _: (),
+) -> Result<U32x2, AnyError>
+where
+  FP: FfiPermissions + 'static,
+{
+  let permissions = state.borrow_mut::<FP>();
+  permissions.check(None)?;
+
+  Ok(U32x2::from(buf.as_ptr() as u64))
+}
+
+fn op_ffi_buf_copy_into<FP>(
+  state: &mut deno_core::OpState,
+  (src, mut dst, len): (U32x2, ZeroCopyBuf, usize),
+  _: (),
+) -> Result<(), AnyError>
+where
+  FP: FfiPermissions + 'static,
+{
+  let permissions = state.borrow_mut::<FP>();
+  permissions.check(None)?;
+
+  let src = u64::from(src) as *const u8;
+  unsafe { ptr::copy(src, dst.as_mut_ptr(), len) };
+  Ok(())
+}
+
+fn op_ffi_cstr_read<FP>(
+  state: &mut deno_core::OpState,
+  ptr: U32x2,
+  _: (),
+) -> Result<String, AnyError>
+where
+  FP: FfiPermissions + 'static,
+{
+  let permissions = state.borrow_mut::<FP>();
+  permissions.check(None)?;
+
+  let ptr = u64::from(ptr) as *const i8;
+  Ok(unsafe { CStr::from_ptr(ptr) }.to_str()?.to_string())
+}
+
+fn op_ffi_read_u8<FP>(
+  state: &mut deno_core::OpState,
+  ptr: U32x2,
+  _: (),
+) -> Result<u8, AnyError>
+where
+  FP: FfiPermissions + 'static,
+{
+  let permissions = state.borrow_mut::<FP>();
+  permissions.check(None)?;
+
+  Ok(unsafe { *(u64::from(ptr) as *const u8) })
+}
+
+fn op_ffi_read_i8<FP>(
+  state: &mut deno_core::OpState,
+  ptr: U32x2,
+  _: (),
+) -> Result<i8, AnyError>
+where
+  FP: FfiPermissions + 'static,
+{
+  let permissions = state.borrow_mut::<FP>();
+  permissions.check(None)?;
+
+  Ok(unsafe { *(u64::from(ptr) as *const i8) })
+}
+
+fn op_ffi_read_u16<FP>(
+  state: &mut deno_core::OpState,
+  ptr: U32x2,
+  _: (),
+) -> Result<u16, AnyError>
+where
+  FP: FfiPermissions + 'static,
+{
+  let permissions = state.borrow_mut::<FP>();
+  permissions.check(None)?;
+
+  Ok(unsafe { *(u64::from(ptr) as *const u16) })
```

Comment:
This won't work on architectures where unaligned loads trap. Use `std::ptr::read_unaligned()` instead.